backend/api/v1/risk_service.go

```diff
@@ -24,22 +27,139 @@ func NewRiskService(store *store.Store) *RiskService {
 	}
 }
 
+func convertToRisk(risk *store.RiskMessage) *v1pb.Risk {
+	return &v1pb.Risk{
+		Name:       fmt.Sprintf("%s%v", riskPrefix, risk.ID),
+		Uid:        fmt.Sprintf("%v", risk.ID),
+		Source:     convertToSource(risk.Source),
+		Title:      risk.Name,
+		Level:      risk.Level,
+		Expression: risk.Expression,
+		Active:     risk.Active,
+	}
+}
+
 // ListRisks lists risks.
-func (*RiskService) ListRisks(context.Context, *v1pb.ListRisksRequest) (*v1pb.ListRisksResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method ListRisks not implemented")
+func (s *RiskService) ListRisks(ctx context.Context, _ *v1pb.ListRisksRequest) (*v1pb.ListRisksResponse, error) {
+	risks, err := s.store.ListRisks(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, err.Error())
+	}
+	response := &v1pb.ListRisksResponse{}
+	for _, risk := range risks {
+		response.Risks = append(response.Risks, convertToRisk(risk))
+	}
+	return response, nil
 }
 
 // CreateRisk creates a risk.
-func (*RiskService) CreateRisk(context.Context, *v1pb.CreateRiskRequest) (*v1pb.Risk, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method CreateRisk not implemented")
+func (s *RiskService) CreateRisk(ctx context.Context, request *v1pb.CreateRiskRequest) (*v1pb.Risk, error) {
+	principalID := ctx.Value(common.PrincipalIDContextKey).(int)
+	risk, err := s.store.CreateRisk(ctx, &store.RiskMessage{
+		Source:     convertSource(request.Risk.Source),
+		Level:      request.Risk.Level,
+		Name:       request.Risk.Title,
+		Active:     request.Risk.Active,
+		Expression: request.Risk.Expression,
+	}, principalID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, err.Error())
+	}
+	return convertToRisk(risk), nil
 }
 
 // UpdateRisk updates a risk.
-func (*RiskService) UpdateRisk(context.Context, *v1pb.UpdateRiskRequest) (*v1pb.Risk, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method UpdateRisk not implemented")
+func (s *RiskService) UpdateRisk(ctx context.Context, request *v1pb.UpdateRiskRequest) (*v1pb.Risk, error) {
+	principalID := ctx.Value(common.PrincipalIDContextKey).(int)
+	if request.UpdateMask == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "update_mask must be set")
+	}
+	riskID, err := getRiskID(request.Risk.Name)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+	risk, err := s.store.GetRisk(ctx, riskID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get risk, error: %v", err)
+	}
+	if risk == nil {
+		return nil, status.Errorf(codes.NotFound, "risk %v not found", request.Risk.Name)
+	}
+	if risk.Deleted {
+		return nil, status.Errorf(codes.InvalidArgument, "risk %v has been deleted", request.Risk.Name)
+	}
+
+	patch := &store.UpdateRiskMessage{}
+	for _, path := range request.UpdateMask.Paths {
+		switch path {
+		case "risk.title":
+			patch.Name = &request.Risk.Title
+		case "risk.active":
```

Comment:
nit: thinking more of it, this should be called risk.status with ACTIVE and INACTIVE enum. Maybe next PR.